src/assisted_test_infra/test_infra/controllers/minikube.py

```diff
@@ -0,0 +1,222 @@
+    _get_obj_resources_by_name -> get the obj resource name
+    """
+    CONFIGMAP = "ConfigMap"
+    DEPLOYMENT = "Deployment"
+    REPLICASET = "ReplicaSet"
+    POD = "Pod"
+    SERVICE = "Service"
+
+    def __init__(self, k8_client):
+        """
+        Initialized k8_client accepted from fixture
+        :param k8_client:
+        """
+        self.k8_client = k8_client
+        # mapping for resources kind
+        self.mapping_resources = {
+            self.CONFIGMAP: self._get_resources_by_kind(self.CONFIGMAP, "v1"),
+            self.DEPLOYMENT: self._get_resources_by_kind(self.DEPLOYMENT, "apps/v1"),
+            self.REPLICASET: self._get_resources_by_kind(self.REPLICASET, "apps/v1"),
+            self.POD: self._get_resources_by_kind(self.POD, "v1"),
+            self.SERVICE: self._get_resources_by_kind(self.SERVICE, "v1")}
+
+    @property
+    def namespace(self):
+        return os.environ.get("NAMESPACE", "assisted-installer")
+
+    @property
+    def configmap_name(self):
+        return "assisted-service-config"
+
+    @property
+    def configmap_data(self):
+        config_map_store = self._get_obj_resources_by_name(self.CONFIGMAP, self.configmap_name)
+        assert len(config_map_store) == 1, "configmap kind name should be unique"
+        return config_map_store[0]
+
+    @property
+    def assisted_service_name(self):
+        return "assisted-service"
+
+    def resource_kind(self, name):
+        """Get the resource kind object by name
+        :param name: ReplicaSet, Deployment, Pod, ConfigMap, Service Pod
+        :return: Resource object
+        """
+        return self.mapping_resources.get(name)
+
+    def _get_resources_by_kind(self, kind_name, api_version):
+        """Get resource by api_version and kind
+        :param kind_name: ReplicaSet, Deployment, Pod, ConfigMap, Service Pod
+        :param api_version: for Pod/service its v1 and others apps/v1
+        :return: resource client based on type Pod , Service others.
+        """
+        return self.k8_client.resources.get(api_version=api_version, kind=kind_name)
+
+    def _get_obj_resources_by_name(self, resource_kind_name, name_begins_with,
+                                   exact_match=False):
+        """Get object by resource obj by namespace and begins name.
+        In case calling to pod resource we may have multiple pods beginning
+        the same name. assisted-service, assisted-service123 ...
+        :param resource_kind_name: ReplicaSet, Deployment, Pod, ConfigMap, Service Pod
+        :param name_begins_with: lookup name begins the same for all kinds
+        :return: List of all objects beginning the same name for pods
+        """
+        all_resources = self.resource_kind(resource_kind_name).get(
+            namespace=self.namespace, name=None)
+        if not exact_match:
+            replicas = [obj for obj in all_resources.items if name_begins_with in obj.metadata.name]
```

Comment:
```suggestion
            replicas = [obj for obj in all_resources.items if metadata.name.startswith(name_begins_with)]
```
to match the name of the variable `name_begins_with` with the behaviour